crates/orchestrator/src/tests/jobs/mod.rs

```diff
@@ -9,14 +7,476 @@ pub mod proving_job;
+
+        // waiting for workers to complete the processing
+        let (result_1, result_2) = tokio::join!(worker_1, worker_2);
+
+        assert_ne!(
+            result_1.unwrap().is_ok(),
+            result_2.unwrap().is_ok(),
+            "One worker should succeed and the other should fail"
+        );
+
+        // Waiting for 5 secs for job to be updated in the db
+        sleep(Duration::from_secs(5)).await;
+
+        let final_job_in_db = db_client.get_job_by_id(job_item.id).await.unwrap().unwrap();
+        assert_eq!(final_job_in_db.status, JobStatus::PendingVerification);
+    }
+
+    /// Tests `verify_job` function when job is having expected status
+    /// and returns a `Verified` verification status.
+    #[rstest]
+    #[tokio::test]
+    async fn verify_job_handler_with_expected_job_status_and_verified_status_return() {
+        let job_item =
+            build_job_item_by_type_and_status(JobType::DataSubmission, JobStatus::PendingVerification, "1".to_string());
+
+        // building config
+        TestConfigBuilder::new().build().await;
+
+        let config = config().await;
+        let database_client = config.database();
+        let mut job_handler = MockJob::new();
+
+        // creating job in database
+        database_client.create_job(job_item.clone()).await.unwrap();
+        // expecting process job function in job processor to return the external ID
+        job_handler.expect_verify_job().times(1).returning(move |_, _| Ok(JobVerificationStatus::Verified));
+        job_handler.expect_max_process_attempts().returning(move || 2u64);
+
+        let job_handler: Arc<Box<dyn Job>> = Arc::new(Box::new(job_handler));
+        let ctx = mock_factory::get_job_handler_context();
+        // Mocking the `get_job_handler` call in create_job function.
+        ctx.expect().times(1).with(eq(JobType::DataSubmission)).returning(move |_| Arc::clone(&job_handler));
+
+        let _ = verify_job(job_item.id).await.is_ok();
+
+        // DB checks.
+        let updated_job = database_client.get_job_by_id(job_item.id).await.unwrap().unwrap();
+        assert_eq!(updated_job.status, JobStatus::Completed);
+
+        // Queue checks.
+        let consumed_messages_verification_queue =
+            config.queue().consume_message_from_queue(JOB_VERIFICATION_QUEUE.to_string()).await.unwrap_err();
+        assert!(matches!(consumed_messages_verification_queue, QueueError::NoData));
+        let consumed_messages_processing_queue =
+            config.queue().consume_message_from_queue(JOB_PROCESSING_QUEUE.to_string()).await.unwrap_err();
+        assert!(matches!(consumed_messages_processing_queue, QueueError::NoData));
+    }
+
+    /// Tests `verify_job` function when job is having expected status
+    /// and returns a `Rejected` verification status.
+    #[rstest]
+    #[tokio::test]
+    async fn verify_job_handler_with_expected_job_status_and_rejected_status_return_and_adds_process_to_job_queue() {
```

Comment:
we can shorten these names a lot, we can just say `verify_job_rejected_status_works`